### PR TITLE
Fix visual bugs in Choice

### PR DESCRIPTION
**Pull request by wishdasher**
* Align the checkbox / radio button with the first line of the label (it was being centered with the entire label before)
* Remove leading spacing of 8 for a group when is no group label, description, or error message
* Remove padding of 1 from `legend` element

* Add an example how to use CheckboxGroup in an existing context that has a different style than the spec

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 87cef50e4ab760b24a2382ddf663def4949a896e

https://deploy-preview-259--wonder-blocks.netlify.com

File: packages/wonder-blocks-form/components/checkbox-group.md [hunk 1]
```diff
@@ -63,3 +60,70 @@ class CheckboxGroupPizzaExample extends React.Component {
     <CheckboxGroupPizzaExample />
 </View>
 ```
+
+This example shows how one can add custom styles to the checkbox group and to
```
Comment (jeresig) [inline packages/wonder-blocks-form/components/checkbox-group.md]: Great example!

File: packages/wonder-blocks-form/components/choice-internal.js [hunk 2]
```diff
@@ -146,16 +148,23 @@ export default class ChoiceInternal extends React.Component<Props> {
...
-        alignItems: "center",
+        alignItems: "flex-start",
         outline: "none",
     },
+    label: {
+        // NOTE: The checkbox/radio button (height 16px) should be center
+        // aligned with the first line of the label. However, LabelMedium has a
+        // declared line height of 20px, so we need to adjust the top to get the
+        // desired alignment.
+        marginTop: -2,
```
Comment (jeresig) [inline packages/wonder-blocks-form/components/choice-internal.js]: 🤷‍♂️ Hmm - this feels a bit awkward, is there any way we could use flexbox to achieve this?

File: packages/wonder-blocks-form/components/radio-group.js [hunk 3]
```diff
@@ -89,9 +89,11 @@ export default class RadioGroup extends React.Component<RadioGroupProps> {
             <StyledFieldset style={styles.fieldset}>
                 {/* We have a View here because fieldset cannot be used with flexbox*/}
                 <View style={style}>
-                    <legend>
-                        <LabelMedium>{label}</LabelMedium>
-                    </legend>
+                    {label && (
+                        <legend style={{padding: 0}}>
```
Comment (jeresig) [inline packages/wonder-blocks-form/components/radio-group.js]: 🛑 Same here with StyleSheet.

File: packages/wonder-blocks-form/components/choice-internal.js (hunk 2 above)
Comment (wishdasher) [reply packages/wonder-blocks-form/components/choice-internal.js]: Not sure -- https://stackoverflow.com/questions/32118013/align-icon-vertically-to-the-center-of-the-first-line-of-text suggests changing the `line-height` of the label, but that's probably not what we want to affect.

Comment (kevinbarabash) [reply packages/wonder-blocks-form/components/choice-internal.js]: Unfortunately it's hard to actually calculate how CSS line-height affects the positioning of the text relative to other things on the line.  See https://iamvdo.me/en/blog/css-font-metrics-line-height-and-vertical-align for the gory details.

File: packages/wonder-blocks-form/components/radio-group.js (hunk 3 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-form/components/radio-group.js]: @wishdasher did this get addressed in a followup PR?

Comment (wishdasher) [reply packages/wonder-blocks-form/components/radio-group.js]: It was addressed in this one!

Comment (kevinbarabash) [reply packages/wonder-blocks-form/components/radio-group.js]: My bad, usually GitHub hides comments on lines that have changed.

**Pull request merged**